### PR TITLE
[dev] gettext: enable subpackages 'libs' & 'common-devel'

### PR DESCRIPTION
**Pull request by mfrw**
- gettext: export 'common-devel' & 'libs' subpackage
- gettext: update manifests

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Exports subpackages 'libs' & 'common-devel'

###### Change Log  

- export subpackage 'libs' & 'common-devel'

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- localbuild

### CODE REVIEW

File: SPECS/gettext/gettext.spec [hunk 1]
```diff
@@ -15,6 +15,24 @@ These allow programs to be compiled with NLS
...
+# autopoint archive
+License: GPLv3+
+BuildArch: noarch
+
+%description common-devel
+This package contains common architecture independent gettext development files.
+
+%package libs
+Summary: Libraries for %{name}
+# libasprintf is LGPLv2+
```
Comment (PawelWMS) [inline SPECS/gettext/gettext.spec:29]: Please use the `License` tag instead - they can be separate for each subpackage.

File: SPECS/gettext/gettext.spec [hunk 2]
```diff
@@ -55,7 +73,19 @@ make %{?_smp_mflags} check
 %{_datarootdir}/%{name}/*
 %{_mandir}/*
 
+%files common-devel
+%{_datadir}/%{name}/archive.*.tar.xz
+
+%files libs
```
Comment (PawelWMS) [inline SPECS/gettext/gettext.spec:79]: Please remove these from the default package now. It seems there are almost no dependencies on `gettext` at the moment, so it is safe to do that (with more dependencies, we might need to leave the duplicates or use `Provides` instead). Ideally, we could also use this opportunity to split out the `*-devel` subpackage, but one thing at a time...;)

File: toolkit/resources/manifests/package/toolchain_aarch64.txt [hunk 3]
```diff
@@ -114,8 +114,8 @@ gdbm-1.18-4.cm2.aarch64.rpm
 gdbm-debuginfo-1.18-4.cm2.aarch64.rpm
 gdbm-devel-1.18-4.cm2.aarch64.rpm
 gdbm-lang-1.18-4.cm2.aarch64.rpm
-gettext-0.21-1.cm2.aarch64.rpm
-gettext-debuginfo-0.21-1.cm2.aarch64.rpm
+gettext-0.21-2.cm2.aarch64.rpm
```
Comment (PawelWMS) [inline toolkit/resources/manifests/package/toolchain_aarch64.txt:117]: Please also update `toolkit/resources/manifests/package/update_manifests.sh`. The new subpackages should be mentioned inside the `toolchain_*txt` manifests but I believe they are unnecessary inside `pkggen_core_*.txt`. Running the script should yield such results. This means that you'll need to update the `remove_packages_for_pkggen_core` function to remove filter out the new subpackages.

File: SPECS/gettext/gettext.spec (hunk 2 above)
Comment (mfrw) [reply SPECS/gettext/gettext.spec]: Sure, it makes compelete sense after our offline discussion.
Appreciate you being vigilant. Thanks

**Review by thcrain-msft — APPROVED** [review]
🚢

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**